proto/src/serializers/bytes.rs

```diff
@@ -106,3 +107,8 @@ pub mod string {
         serializer.serialize_str(&string)
     }
 }
+
+/// Helper struct to serialize/deserialize Base64-encoded strings into Vec<u8> bytestream.
+#[derive(Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct Base64(#[serde(with = "crate::serializers::bytes::base64string")] pub Vec<u8>);
```

Comment:
Do you anticipate reusing this struct elsewhere?

Reply:
It's a helper struct so the Data struct can do validation. If I implement it directly on Data, then the validation will not be called.

It can be reused for any other Base64-encoding where there is also validation involved. I expect that these helper structs will end up in the JSON create, eventually.

Reply:
Actually, since the Data structure doesn't have any validation now, we can squash this and the Data struct. I'm just worried keeping serialization on domain types. As Ethan put it "friends don't let friends serialize primitive types directly".

I'd keep it but if you want less code, we can remove it.

Reply:
I'm personally a fan of minimizing the amount of code that we write, but not at the expense of readability. So if it makes the code easier to understand, I'd say keep it. If not, I'd say ditch it 😁

Reply:
done, ditched it